src/yarn/typescript-workspace.ts

```diff
@@ -4,15 +4,57 @@ import { Component, javascript, typescript, TaskStep, SourceCode, DependencyType
 import { Monorepo } from './monorepo';
 import { TypeScriptWorkspaceOptions } from './typescript-workspace-options';
 
+
+/**
+ * A reference to a workspace in the same monorepo
+ *
+ * This can be used to reference packages in the same monorepo when declaring
+ * `deps` and `devDeps`.
+ *
+ * By default, all `TypeScriptWorkspace`s implement this interface, representing themselves
+ * with a `^` dependency and a restriction that public packages must have public dependencies.
+ *
+ * The method `.customizeReference({ ... })` can be used to create a workspace reference with
+ * different behavior.
+ */
+export interface IWorkspaceReference {
+  /**
+   * Whether the referenced workspace package is private
+   */
+  readonly isPrivatePackage: boolean;
+
+  /**
+   * Whether we should allow a dependency on this package, even if it is private.
+   */
+  readonly allowPrivateDependency: boolean;
```

Comment:
This feels like the wrong way around

Reply:
Tell me?

Reply:
A (source) package depends on a workspace reference, and the workspace reference decide if it can be dependet on despite being private, but the (source) package needs to ensure the workspace ref is bundled.

Reply:
Feels like the source package should just have a flag to bypass the guardrail à la "stop babysitting me, I know what I'm doing".